docs/content/guides/developer/advanced/custom-indexer/indexer-data-integration.mdx

```diff
@@ -57,6 +61,35 @@ $ cargo run -- --rpc-api-url https://fullnode.testnet.sui.io:443
 - Networks without official remote store (devnet, localnet, custom networks)
 - Development against local Sui networks
 
+### Push-based data sources
+
+Push-based sources provide real-time checkpoint data as it becomes available, offering lower latency than polling.
+
+#### gRPC streaming
+
+gRPC streaming provides real-time checkpoint data pushed from fullnodes for the latest checkpoints. Because streaming only provides the latest data, it requires a polling-based fallback source (remote store, local path, or RPC API) for historical checkpoints and reliability:
```

Comment:
```suggestion
gRPC streaming delivers real-time checkpoint data pushed from full nodes for the latest checkpoints. Because it only streams latest data, you must configure a polling-based fallback source (remote store, local path, or full node RPC) to retrieve historical checkpoints and ensure reliability:
```